jobsdb/jobsdb.go

```diff
@@ -2058,22 +2058,15 @@ func (jd *HandleT) backupTable(backupDSRange dataSetRangeT, isJobStatusTable boo
 			panic(err)
 		}
 
-		var rows []interface{}
-		err = json.Unmarshal(rawJSONRows, &rows)
-		if err != nil {
-			panic(err)
-		}
-		contentSlice := make([][]byte, len(rows))
-		for idx, row := range rows {
-			rowBytes, err := json.Marshal(row)
-			if err != nil {
-				panic(err)
-			}
-			contentSlice[idx] = rowBytes
+		if len(rawJSONRows) < 2 {
+			break
 		}
-		// append new line character at end, before next backupRowsBatch is appended to same file
-		content := append(bytes.Join(contentSlice[:], []byte("\n")), []byte("\n")...)
-		gzWriter.Write(content)
+
+		rawJSONRows = bytes.Replace(rawJSONRows, []byte(", \n "), []byte("\n"), -1) //replacing ", \n " with "\n"
```

Comment:
This feels little risky to me, as we might end up with malformed json. Let's add an assert for number of rows equals number of replacements? I have checked current schemas, and as we are using jsonb types for json, any newlines in the data must be stripped off already. But in case ", \n" happens in some data, it might cause issue.

Reply:
assert added.